bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadataBuilder.java

```diff
@@ -73,14 +73,9 @@ public static LedgerMetadataBuilder from(LedgerMetadata other) {
         builder.ackQuorumSize = other.getAckQuorumSize();
 
         builder.state = other.getState();
-
-        long lastEntryId = other.getLastEntryId();
-        if (lastEntryId != LedgerHandle.INVALID_ENTRY_ID) {
-            builder.lastEntryId = Optional.of(lastEntryId);
-        }
-        long length = other.getLength();
-        if (length > 0) {
-            builder.length = Optional.of(length);
+        if (builder.state == LedgerMetadataFormat.State.CLOSED) {
```

Comment:
Is this a behavior change?
Do you think it is worth to add a comment?

Reply:
The previous behaviour was incorrect. length should never be set if the state is anything but closed. I don't think it needs a comment. the intent is clear.